test/cookies.ts

```diff
@@ -14,7 +15,7 @@ test('reads a cookie', withServer, async (t, server, got) => {
 
 	const cookieJar = new toughCookie.CookieJar();
 
-	await got({cookieJar});
+	await got({cookieJar} as unknown as OptionsOfDefaultResponseBody);
```

Comment:
Why does this need force-casting?

Reply:
I did this because I hit an issue where TypeScript is always using the first overload of the CookieJar's methods to match the use cases in our source code, which is actually using the third overload. Is there a way to make a proper comparison happen?

Reply:
Overloads are order-dependent, so maybe try changing the order? If this is in fact a TS issue, it would be good to try to find an issue on the TS issue tracker and add a code comment linking to it.